include/api/CAnomalyJobConfig.h

```diff
@@ -234,7 +257,7 @@ class API_EXPORT CAnomalyJobConfig {
 
     std::string jobId() const { return m_JobId; }
     std::string jobType() const { return m_JobType; }
-    const CAnalysisConfig& analysisConfig() const { return m_AnalysisConfig; }
```

Comment:
You could provide _both_ const and non-const accessors.  In future as this class gets used more widely some places may hold a const reference to the `CAnomalyJobConfig`, and it's annoying for those places to have to cast away constness just to get a const reference to one of the sub-objects.